src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -60,6 +62,17 @@ instance (KnownNat n, KnownNat l, Arithmetic (ScalarField c1), Arbitrary (Scalar
         let (omega, k1, k2) = getParams (value @n)
         Plonk omega k1 k2 (Vector vecPubInp) ac <$> arbitrary
 
+instance forall n l c1 c2 t . (KnownNat n, KnownNat l, Eq (ScalarField c1), Arithmetic (ScalarField c1), Arbitrary (ScalarField c1),
```

Comment:
`Eq a` is a superclass of `Arithmetic a`, so this looks like a redundant constraint
Also it's strange that `Witness` type family doesn't compute, what does it say if one removes a constraint on witness?

Reply:
It says "could not deduce `Witness (Plonk n l c1 c2 t) ~ (PlonkWitnessInput c0, PlonkProverSecret b0)`